### PR TITLE
Adds RHEL based Dockerfile for client kn

### PR DESCRIPTION
**Pull request by navidshaikh**
as per title

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN TAG="v0.8.0" make build
+
+FROM ubi8:8-released
+COPY --from=builder /opt/app-root/src/go/src/github.com/knative/client/kn /usr/bin/kn
+
+LABEL \
+      com.redhat.component="openshift-serverless-1-tech-preview-client-kn-rhel8-container" \
+      name="openshift-serverless-1-tech-preview/client-kn-rhel8" \
+      version="0.8.0" \
+      summary="Red Hat OpenShift Serverless 1 Client kn container" \
```
Comment (rhuss) [inline Dockerfile]:
```suggestion
      summary="Red Hat OpenShift Serverless 1 - kn CLI" \
```
Shouldn't we add also 'cli' to the summary and/or description ?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM rhel8/go-toolset:1.12.8 AS builder
+WORKDIR /opt/app-root/src/go/src/github.com/knative/client
+COPY . .
+RUN TAG="v0.8.0" make build
```
[the remark this answers is not in the document]
Comment (navidshaikh) [reply Dockerfile]: > Not using v0.8.1?


will have `v0.8.0`

[the remark this answers is not in the document]
Comment (navidshaikh) [reply Dockerfile]: > go build knative.dev/client/cmd/kn: open kn: permission denied

this is interesting - thanks for giving it a try Warren, I will try that.

**Comment by navidshaikh** [comment]
> I see the following build failure when running docker build locally. Not sure if it's an issue with rhpkg build.

@warrenvw 
You are right! I see this issue locally as well using rhpkg. While running the build using the default user `default`  in builder image, we get
```
./hack/build.sh -f
🚧 Compile
go build knative.dev/client/cmd/kn: open kn: permission denied
make: *** [Makefile:25: build] Error 1
```
Using `USER root` in builder image solves this. :thinking:

**Comment by navidshaikh** [comment]
The WORKDIR created under builder image has user and owner set as `1000` and `1000`.
The WORKDIR and its contents permissions look like `drwxrwxr-x.`.

While the builder image default user `default` executing the `go build` has uid `1001` and gid `0`.
Meaning, it belongs to `others` (for WORKDIR) which has only read and execute `r-x`, not write permission to export the kn binary in same WORKDIR, thus permission issue.

We need a provision to specify the output path for kn binary in makefile while leveraging the ldflags.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+COPY --from=builder /opt/app-root/src/go/src/github.com/knative/client/kn /usr/bin/kn
+
+LABEL \
+      com.redhat.component="openshift-serverless-1-tech-preview-client-kn-rhel8-container" \
+      name="openshift-serverless-1-tech-preview/client-kn-rhel8" \
+      version="0.8.0" \
+      summary="Red Hat OpenShift Serverless 1 - CLI client kn" \
+      description="CLI client 'kn' for managing Red Hat OpenShift Serverless 1" \
+      io.k8s.display-name="Red Hat OpenShift Serverless 1 - CLI client kn" \
+      io.k8s.description="CLI client 'kn' for managing Red Hat OpenShift Serverless 1" \
```
Comment (warrenvw) [inline Dockerfile]: I would like to see summary and display-name values replaced with something more consistent to what we use for Knative Serving. Perhaps "Red Hat OpenShift Serverless 1 Kn CLI"... or if you feel 'client' must be in the name, "Red Hat OpenShift Serverless 1 Kn Client CLI"?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+FROM ubi8:8-released
+COPY --from=builder /opt/app-root/src/go/src/github.com/knative/client/kn /usr/bin/kn
+
+LABEL \
+      com.redhat.component="openshift-serverless-1-tech-preview-client-kn-rhel8-container" \
+      name="openshift-serverless-1-tech-preview/client-kn-rhel8" \
+      version="0.8.0" \
+      summary="Red Hat OpenShift Serverless 1 Client kn container" \
+      description="Red Hat OpenShift Serverless 1 Client kn container" \
+      io.k8s.description="Red Hat OpenShift Serverless 1 Client kn container" \
```
[the remark this answers is not in the document]
Comment (rhuss) [reply Dockerfile]: My question is, if `summary` and `description` are always the same what value does it have to have two different labels ?

File: Dockerfile (hunk 3 above)
Comment (rhuss) [reply Dockerfile]: I could live with "Red Hat OpenShift Serverless 1 kn CLI" (but please with lowercase 'k' in 'kn' as this is how we refer to it in the documentation and how it is called on the command line).

Comment (navidshaikh) [reply Dockerfile]: Changed to `Red Hat OpenShift Serverless 1 kn CLI`.

**Pull request merged**